public/modules/custom/grants_attachments/src/AttachmentFixerService.php

```diff
@@ -0,0 +1,251 @@
+        $innerArray[] = [
+          'ID' => 'integrationID',
+          'label' => 'Integration ID',
+          'value' => $intID,
+          'valueType' => 'string',
+          'meta' => "[]",
+        ];
+      }
+    }
+  }
+
+  /**
+   * Try to figure our if attachments are ok.
+   *
+   * This is a helper function for fixAttachmentsOnApplication that goes through
+   * the events and attachment info to see if the attachments are added properly
+   * to the document.
+   *
+   * @param mixed $events
+   *   Events.
+   * @param mixed $attachment
+   *   Attachment.
+   * @param mixed $attachmentInfo
+   *   Attachment info.
+   * @param mixed $appEnv
+   *   Application environment.
+   *
+   * @return array
+   *   Array of booleans.
+   */
+  public function areAttachmentsOk(
+    mixed $events,
+    mixed $attachment,
+    mixed $attachmentInfo,
+    mixed $appEnv,
+  ): array {
+    // Look for events from us, the handler.
+    $handlerOk = $this->filterEventsByTypeAndFilename($events, 'HANDLER_ATT_OK', $attachment['filename']);
+    // Look for events from AVUSTUS2.
+    $avus2Ok = $this->filterEventsByTypeAndFilename($events, 'AVUSTUS2_ATT_OK', $attachment['filename']);
+    // Look for errors from AVUSTUS2.
+    $avus2Error = $this->filterEventsByTypeAndFilename($events, 'AVUSTUS2_ATT_ERROR', $attachment['filename']);
+    // Are attachments added properly to form data.
+    $formOk = $this->checkAttachmentInfo($attachmentInfo, $attachment, $appEnv);
+
+    return [
+      'handler' => !empty($handlerOk),
+      'avus2' => !empty($avus2Ok),
+      'avus2Errors' => $avus2Error,
+      'form' => !empty($formOk),
+    ];
+  }
+
+  /**
+   * Filter events.
+   *
+   * @param array $events
+   *   Events.
+   * @param string $eventType
+   *   Event type.
+   * @param string $filename
+   *   Filename.
+   *
+   * @return array
+   *   Filtered events.
+   */
+  private function filterEventsByTypeAndFilename(array $events, string $eventType, string $filename): array {
+    return array_filter($events, function ($event) use ($eventType, $filename) {
+      return $event['eventType'] === $eventType && $event['eventTarget'] === $filename;
+    });
+  }
+
+  /**
+   * Check attachment info for existing intergation ID.
+   *
+   * @param mixed $attachmentInfo
+   *   Attachment info.
+   * @param mixed $attachment
+   *   Attachment.
+   * @param mixed $appEnv
+   *   Application environment.
+   *
+   * @return bool
+   *   True if found, false otherwise.
+   */
+  private function checkAttachmentInfo(mixed $attachmentInfo, mixed $attachment, mixed $appEnv): bool {
+    // If no attachments, nothing to check.
+    if (!is_array($attachmentInfo)) {
+      return FALSE;
+    }
+
+    // Loop attachment info and check if we have the attachment added.
+    foreach ($attachmentInfo as $info) {
+      // Check if filename and integrationID are found.
+      $filenameFound = $this->findValueById($info, 'fileName', $attachment['filename']);
```

Comment:
`findValueById` expects `$info` and `$attachment` to be an array, but this method allows `mixed` type for both. The type should be enforced before calling the method.